crates/uv/tests/common/mod.rs

```diff
@@ -487,6 +491,31 @@ impl TestContext {
             &format!("{}{}", self.python_kind(), self.python_version),
         )
     }
+
+    /// Reset the virtual environment in the test context.
+    pub fn reset_venv(&self) {
+        create_venv_from_executable(
+            &self.temp_dir,
+            &self.cache_dir,
+            &get_toolchain(&self.python_version),
+        );
+    }
+
+    /// Create a new virtual environment named `.venv` in the test context.
+    fn create_venv(&mut self, python: &str) -> PathBuf {
+        let parent = self.temp_dir.to_path_buf();
+        self.create_venv_in_parent(parent, python)
+    }
+
+    /// Create a new virtual environment named `.venv` in the given directory.
+    fn create_venv_in_parent<P: AsRef<Path>>(&mut self, path: P, python: &str) -> PathBuf {
+        let executable = get_toolchain(python);
+        self.add_filters_for_python_version(
+            &PythonVersion::from_str(python).unwrap(),
+            executable.clone(),
+        );
+        create_venv_from_executable(&ChildPath::new(path.as_ref()), &self.cache_dir, &executable)
+    }
```

Comment:
I moved these here then noticed we don't actually need them in tests after adding `reset` so I made them private.